code/script.js

```diff
@@ -222,104 +228,152 @@ const generateBoard = () => {
+      )
+    } else {
+      alert(
+        `No, the person doesn´t wear ${value}! Remove all people who wear ${value}`
+      )
+    }
+  } else if (category === 'hair') {
+    if (keep) {
+      alert(`Yes, the person has ${value} hair! Keep all people who have ${value} hair`
+      )
     } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+      alert(
+        `No, the person doesn´t have ${value} hair! Remove all people who have ${value} hair`
+      )
+    }
+  } else if (category === 'eyes') {
+    if (keep) {
+      alert(`Yes, the person has ${value} eyes! Keep all people who have ${value} eyes`
+      )
+    } else {
+      alert(
+        `No, the person doesn´t have ${value} eyes! Remove all people who have ${value} eyes`
+      )
     }
   }
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
+  // FILTER BY CATEGORY TO KEEP/REMOVE BASED ON THE KEEP VARIABLE//
+  if (category === 'hair' || category === 'eyes') {
+    charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value) === keep);
+  } else if (category === 'accessories' || category === 'other') {
+    charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value) === keep);
+  }
 
   // Invoke a function to redraw the board with the remaining people.
+  generateBoard()
 }
 
-// when clicking guess, the player first have to confirm that they want to make a guess.
+//CLICK GUESS//
+// when clicking guess, the player has to confirm that they want to make a guess.
 const guess = (personToConfirm) => {
   // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+  const confirmGuess = confirm(`Are you sure you want to guess ${personToConfirm}?`);
+  if (confirmGuess) checkMyGuess(personToConfirm);
 }
 
-// If you confirm, this function is invoked
+//WIN OR LOSE.//
+// If they confirm, this function is invoked
 const checkMyGuess = (personToCheck) => {
-  // 1. Check if the personToCheck is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
+  if (personToCheck === secret.name) {
+    winOrLoseText.innerHTML = `That's correct!!! You won! Congratulations!`;
+  } else {
+    personToConfirm !== secret.name;
```

Comment:
In the checkMyGuess function, you have an unnecessary line: personToConfirm !== secret.name;. It doesn't serve any purpose and can be removed.